### PR TITLE
docker scheme: working-directory and stdin passing

### PR DESCRIPTION
**Pull request by drzraf**
1. `docker exec` and `docker-compose exec` provide an option to cleanly set the working directory before running a command
=> Use it instead and use `cd <...> ;` command-prefixing in vagrant/ssh cases

2. Using `docker` scheme, `wp post update 1 - <<<foo` didn't worked (issue #4972)
* Because STDOUT being a tty, `-t` is passed with no ability to inhibit this
* Because docker `-i` flag is necessary

=> As such:
* docker `-i` (`--interactive`) is introduced (with sensible autodetection)
* `WP_CLI_DOCKER_NO_TTY` and `WP_CLI_DOCKER_NO_INTERACTIVE` environment variables are introduced to optionally inhibit these respective flags

### CODE REVIEW

**Comment by drzraf** [comment]
ping?

**Comment by schlessera** [comment]
I already tested this in the past, and while it seemed to do what it is supposed to do, I was unsure at the time whether I considered all implications and tested all edge cases.
However, this has waiting for long enough already. Let's merge it as-is, I'm sure the edge cases will show themselves on their own accord then... ;)

Thanks for the PR, @drzraf !

**Review by schlessera — APPROVED** [review]

**Comment by swissspidy** [comment]
@schlessera This actually had some test failures, see https://github.com/wp-cli/wp-cli/actions/runs/13191576733/job/36825712748#step:11:295 for example

**Comment by drzraf** [comment]
Can't reproduce the failure locally: `composer behat -- features/flags.feature` passes
```
16 scenarios (16 passed)
98 steps (98 passed)
0m27.49s (11.38Mb)
```

and the full run gives
```
233 scenarios (228 passed, 5 failed)
2416 steps (2390 passed, 5 failed, 21 skipped)
11m39.65s (11.97Mb)
```

but none of these 5 failures matches those mentioned by the CI

**Comment by mrsdizzie** [comment]
The problem is that in the CI it is setting the -i flag (from the new code in this PR), which you can see in the failed test:
```
  Debug (bootstrap): Running SSH command: docker exec --user 'user' -i 'wordpress' sh -c 'wp '\''--debug'\'' '\''--version'\''' (0.074s)
```
But the test does not expect that new `-i`:
https://github.com/wp-cli/wp-cli/blob/63115a0c31e91beab7368888e5c11154afe8ef55/features/flags.feature#L357-L362

The tests pass locally because 
`$is_stdin_tty  = function_exists( 'posix_isatty' ) ? posix_isatty( STDIN ) : true;`
Will return true because STDIN is attached to your local terminal which is a TTY.

You can reproduce the CI failure by piping the test through cat which will remove the tty:
```
composer behat -- features/flags.feature:357 | cat
```

FWIW I think the double negative here is a bit confusing and hard to follow
```
! $is_stdin_tty && ! getenv( 'WP_CLI_DOCKER_NO_INTERACTIVE' ) ? '-i ' : ''
```
The test should probably be updated and include the environment variables so the output is the same across local and ci testing. I think otherwise it would require creating a new custom step to check for tty if we really wanted to test the tty detection as well.
